setup.go

```diff
@@ -29,6 +30,14 @@ func setup(c *caddy.Controller) error {
 	}
 	device := c.Val()
 
+	// parse optional local ip
+	var wgIP net.IP
+	if c.NextArg() {
+		wgIP = net.ParseIP(c.Val())
+	} else {
+		wgIP = getOutboundIP()
```

Comment:
I think we can use `w.LocalAddr()` + `net.SplitHostPort` to get the local address for the socket handing the DNS request